demo/demo.sh

```diff
@@ -14,14 +14,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cd $(dirname ${BASH_SOURCE})
-. ./udemo.sh
-. ../scripts/util.sh
+# Resolve the absolute path to this script directory.
+cd $(dirname "${BASH_SOURCE[0]}")
+script_dir="$(pwd)"
```

Comment:
Given that there is a `scripts` directory, but this points to `demo`, it would be less confusing to name this `demo_dir`. It would also be useful to have a pointer to the project root, or the `scripts` directory:

```suggestion
demo_dir=$(realpath "$(pwd)")
scripts_dir=$(realpath "$(pwd)/../scripts")
```

(assignments from command substitutions don’t need to be quoted).